lib/automation.py

```diff
@@ -361,31 +388,80 @@ def process_build(self, **pulse_properties):
         pulse_properties['build_url'] = self.get_installer_url(pulse_properties)
         pulse_properties['test_packages_url'] = self.get_test_packages_url(pulse_properties)
 
+        # mozharness retrieval has to be last because it relies on the test packages url
```

Comment:
If you want, you can make that dependency explicity by defining `get_mozharness_url` to accept the test packages url as a parameter (instead of passing in `pulse_properties`)